drivers/mdio/mdio_renesas_ra.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/mdio.h>
+#include <zephyr/net/ethernet.h>
+#include <zephyr/net/mdio.h>
+#include "r_ether_phy.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(renesas_ra_mdio, CONFIG_MDIO_LOG_LEVEL);
+
+#define DT_DRV_COMPAT renesas_ra_mdio
+
+struct renesas_ra_mdio_config {
+	const struct pinctrl_dev_config *pincfg;
+	uint8_t instance;
+};
+
+struct renesas_ra_mdio_data {
+	struct k_mutex rw_mutex;
+	struct st_ether_phy_cfg ether_phy_cfg;
+	struct st_ether_phy_instance_ctrl ether_phy_ctrl;
+};
+
+static int renesas_ra_mdio_read(const struct device *dev, uint8_t prtad, uint8_t regad,
+				uint16_t *data)
+{
+	struct renesas_ra_mdio_data *dev_data = dev->data;
+	uint32_t read;
+	fsp_err_t err;
+
+	dev_data->ether_phy_ctrl.phy_lsi_address = prtad;
+
+	k_mutex_lock(&dev_data->rw_mutex, K_FOREVER);
+
+	err = R_ETHER_PHY_Read(&dev_data->ether_phy_ctrl, regad, &read);
+
+	k_mutex_unlock(&dev_data->rw_mutex);
+
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+
+	*data = read & UINT16_MAX;
+
+	return 0;
+}
+
+static int renesas_ra_mdio_write(const struct device *dev, uint8_t prtad, uint8_t regad,
+				 uint16_t data)
+{
+	struct renesas_ra_mdio_data *dev_data = dev->data;
+	fsp_err_t err;
+
+	dev_data->ether_phy_ctrl.phy_lsi_address = prtad;
+
+	k_mutex_lock(&dev_data->rw_mutex, K_FOREVER);
+
+	err = R_ETHER_PHY_Write(&dev_data->ether_phy_ctrl, regad, data);
+
+	k_mutex_unlock(&dev_data->rw_mutex);
+
+	if (err != FSP_SUCCESS) {
+		return -EIO;
+	}
+
+	return 0;
+}
+
+static int renesas_ra_mdio_initialize(const struct device *dev)
+{
+	struct renesas_ra_mdio_data *data = dev->data;
+	const struct renesas_ra_mdio_config *cfg = dev->config;
+	int err;
+	fsp_err_t fsp_err;
+
+	err = pinctrl_apply_state(cfg->pincfg, PINCTRL_STATE_DEFAULT);
+	if (err != 0) {
+		return err;
+	}
+
+	fsp_err = R_ETHER_PHY_Open(&data->ether_phy_ctrl, &data->ether_phy_cfg);
+
+	if (fsp_err != FSP_SUCCESS) {
+		LOG_ERR("Failed to init mdio driver - R_ETHER_PHY_Open fail");
+	}
+
+	k_mutex_init(&data->rw_mutex);
+
+	return 0;
+}
+
+static const struct mdio_driver_api renesas_ra_mdio_api = {
```

Comment:
```suggestion
static DEVICE_API(mdio, renesas_ra_mdio_api) = {
```
like here: https://github.com/zephyrproject-rtos/zephyr/pull/82267

Reply:
Thank you, I rebase and change to use the `DEVICE_API`